Tests/PenguinStructuresTests/AnyValueTests.swift

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinTesting
+import PenguinStructures
+import XCTest
+
+fileprivate protocol P {}
+extension Double: P {}
+
+fileprivate class Base {}
+fileprivate class Derived: Base {}
+
+final class AnyValueTests: XCTestCase {
+  // A type that will not fit in existential inline storage.
+  typealias BufferOverflow = (Int, Int, Int, Int)
+
+  // A value that will not fit in existential inline storage.
+  let bufferOverflow = (1, 2, 3, 4)
+  
+  func test_Int() {
+    var a = AnyValue(3)
+    XCTAssert(a.storedType == Int.self)
+    XCTAssertEqual(a[Type<Int>()], 3)
+    XCTAssertEqual(a[unsafelyAssuming: Type<Int>()], 3)
+    
+    a[Type<Int>()] += 1
+    XCTAssertEqual(a[Type<Int>()], 4)
+    XCTAssertEqual(a.boxOrObjectID, nil)
+    
+    a[unsafelyAssuming: Type<Int>()] += 1
+    XCTAssertEqual(a[Type<Int>()], 5)
+  }
+  
+  func test_String() {
+    var a = AnyValue("Three")
+    XCTAssert(a.storedType == String.self)
+    XCTAssertEqual(a[Type<String>()], "Three")
+    XCTAssertEqual(a[unsafelyAssuming: Type<String>()], "Three")
+
+    a[Type<String>()].append("D")
+    XCTAssertEqual(a[Type<String>()], "ThreeD")
+
+    // I don't know how big String is off the top of my head.
+    let inlineStorageExpected = MemoryLayout<String>.size < 3 * MemoryLayout<Int>.size
+      && MemoryLayout<String>.alignment <= MemoryLayout<Any>.alignment
+    
+    if inlineStorageExpected {
+      XCTAssertEqual(a.boxOrObjectID, nil)
+    }
+    else {
```

Comment:
```suggestion
    } else {
```

Reply:
I'll do this here because it's just a test and why argue, but IMO this formatting pattern generally makes code harder to read, especially because it leads to 
```
  complexCodeAbove
} else if someLongDenseConditionThatBlendsIntoTheBlocksAboveAndBelow {
  complexCodeBelow
```

and even more especially with 2-space indents.